dagshub/data_engine/model/datasource.py

```diff
@@ -61,13 +62,38 @@ class DatapointMetadataUpdateEntry(json.JSONEncoder):
     allowMultiple: bool = False
 
 
+@dataclass
+class Field:
+    column: str
+    as_of_time: Optional[Union[float, datetime.datetime]] = None
+    alias: Optional[str] = None
+
+    @property
+    def as_of_timestamp(self) -> Optional[int]:
+        if self.as_of_time:
+            if isinstance(self.as_of_time, datetime.datetime):
+                return int(self.as_of_time.timestamp())
+            else:
+                return int(self.as_of_time)
+        else:
+            return None
+
+    def to_dict(self) -> Dict[str, Any]:
+        res_dict = {"name": self.column}
+        if self.as_of_time:
+            res_dict["asOf"] = self.as_of_timestamp
```

Comment:
Filtering from the 0 timestamp should still be valid, even if it doesn't make sense
```suggestion
        if self.as_of_time is not None:
            res_dict["asOf"] = self.as_of_timestamp
```